cmd/dsse/dsse.go

```diff
@@ -0,0 +1,319 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return &envelope, nil
+}
+
+func getPAE(envelope *gitrelease.Envelope) ([]byte, error) {
+	payload, err := base64.StdEncoding.DecodeString(envelope.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	preAuthenticationEncoding, err := gitrelease.PreAuthenticationEncoding(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	return preAuthenticationEncoding, nil
+}
+
+func signAndWrite(DSSEPath string, envelope *gitrelease.Envelope, keyfilePath string) error {
+	existing, err := base64.StdEncoding.DecodeString(envelope.Payload)
+	if err != nil {
+		return err
+	}
+
+	err = printPayload(envelope)
+	if err != nil {
+		return err
+	}
+
+	signature, err := sign(existing, keyfilePath)
+	if err != nil {
+		return err
+	}
+
+	envelope.Signatures = append(envelope.Signatures, gitrelease.Signature{
+		Sig: signature,
+	})
+
+	output, err := json.Marshal(envelope)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(DSSEPath, output, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func printPayload(envelope *gitrelease.Envelope) error {
+	payload, err := base64.StdEncoding.DecodeString(envelope.Payload)
+	if err != nil {
+		return err
+	}
+	var t any
+	err = json.Unmarshal(payload, &t)
+	if err != nil {
+		return err
+	}
+
+	formatted, err := json.MarshalIndent(t, "  ", "  ")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(formatted))
+
+	return nil
+}
+
+func sign(payload []byte, keyfilePath string) (string, error) {
+	preAuthenticationEncoding, err := gitrelease.PreAuthenticationEncoding(payload)
+	if err != nil {
+		return "", fmt.Errorf("unable to create signature payload: %w", err)
+	}
+
+	tmpFile, err := os.CreateTemp("", "payload")
+	if err != nil {
+		return "", fmt.Errorf("unable to create temporary file: %w", err)
+	}
+
+	_, err = tmpFile.Write(preAuthenticationEncoding)
+	if err != nil {
+		return "", fmt.Errorf("unable to write to temporary file: %w", err)
+	}
+	err = tmpFile.Sync()
+	if err != nil {
+		return "", fmt.Errorf("unable to sync temporary file: %w", err)
+	}
+
+	command := []string{"ssh-keygen", "-Y", "sign", "-n", sshNamespace, "-f", keyfilePath, tmpFile.Name()}
+
+	fmt.Printf("Executing command: %s\n", command)
+	cmd := exec.Command(command[0], command[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	err = cmd.Run()
+	if err != nil {
+		return "", err
+	}
+
+	signatureFile := tmpFile.Name() + ".sig"
+	sig, err := os.ReadFile(signatureFile)
+	if err != nil {
+		return "", fmt.Errorf("unable to read signature from %s: %w", signatureFile, err)
+	}
+
+	signature := base64.StdEncoding.EncodeToString(sig)
+
+	err = os.Remove(tmpFile.Name())
+	if err != nil {
+		return "", fmt.Errorf("unable to delete temporary file %s: %w", tmpFile.Name(), err)
+	}
+
+	err = os.Remove(tmpFile.Name() + ".sig")
+	if err != nil {
+		return "", fmt.Errorf("unable to delete temporary signature %s: %w", signatureFile, err)
+	}
+
+	return signature, nil
+}
```

Comment:
Remove these with deferred cleanup.
```suggestion
	signature := base64.StdEncoding.EncodeToString(sig)

	return signature, nil
}
```